graviti/openapi/dataset.py

```diff
@@ -0,0 +1,217 @@
+           "default_branch": "main",
+           "commit_id": "a0d4065872f245e4ad1d0d1186e3d397",
+           "cover_url": "https://tutu.s3.cn-northwest-1.amazonaws.com.cn/",
+           "created_at": "2021-03-03T18:58:10Z",
+           "updated_at": "2021-03-03T18:58:10Z",
+           "owner": "czhual",
+           "is_public": false,
+           "config_name": "exampleConfigName"
+        }
+
+    """
+    url = urljoin(url, f"{URL_PATH_PREFIX}/datasets/{dataset_name}")
+    return open_api_do("GET", access_key, url).json()  # type: ignore[no-any-return]
+
+
+def list_datasets(
+    access_key: str,
+    url: str,
+    *,
+    offset: int = 0,
+    limit: int = 128,
+) -> Dict[str, Any]:
+    """Execute the OpenAPI `GET /v1/datasets`.
+
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        offset: The offset of the page.
+        limit: The limit of the page.
+
+    Returns:
+        The response of OpenAPI.
+
+    Examples:
+        >>> list_datasets("ACCESSKEY-********", "https://graviti.com/")
+        {
+           "datasets": [
+               {
+                   "id": "2bc95d506db2401b898067f1045d7f68",
+                   "name": "OxfordIIITPet",
+                   "alias": "Oxford-IIIT Pet",
+                   "default_branch": "main",
+                   "commit_id": "a0d4065872f245e4ad1d0d1186e3d397",
+                   "cover_url": "https://tutu.s3.cn-northwest-1.amazonaws.com.cn/",
+                   "created_at": "2021-03-03T18:58:10Z",
+                   "updated_at": "2021-03-03T18:58:10Z",
+                   "owner": "czhual",
+                   "is_public": false,
+                   "config_name": "exampleConfigName"
+               }
+            ],
+           "offset": 0,
+           "record_size": 1,
+           "total_count": 1
+        }
+
+    """
+    url = urljoin(url, f"{URL_PATH_PREFIX}/datasets")
+    params: Dict[str, Any] = {"offset": offset, "limit": limit}
+
+    return open_api_do("GET", access_key, url, params=params).json()  # type: ignore[no-any-return]
+
+
+def update_dataset(
+    access_key: str,
+    url: str,
+    dataset_name: str,
+    *,
+    name: Optional[str] = None,
+    alias: Optional[str] = None,
+    is_public: Optional[bool] = None,
+    default_branch: Optional[str] = None,
+) -> None:
+    """Execute the OpenAPI `PATCH /v1/datasets/{dataset_name}`.
+
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        dataset_name: Name of the dataset, unique for a user.
+        name: New name of the dataset, unique for a user.
+        alias: New alias of the dataset.
+        is_public: Whether the dataset is public.
+        default_branch: User's chosen branch.
+
+    Examples:
+        >>> update_dataset(
+        ...     "ACCESSKEY-********",
+        ...     "https://graviti.com/",
+        ...     "OxfordIIITPet",
+        ...     name="OxfordIIIT Pet",
```

Comment:
we cannot add space in dataset name, because  it will be added to the URL.
Here, we could use "OxfordIIITCat" and spaces is allowed in alias, so it could be "OxfordIIIT Cat"